src/lib.rs

```diff
@@ -86,6 +88,29 @@ pub fn run(
 
     log::info!("Loading book summary: {}", mdbook_summary_path.display());
     let summary_src = std::fs::read_to_string(&mdbook_summary_path)?;
+
+    // Filter out any slides given in the `skip_slides` toml array entry
+    let summary_src = match book_skip_slides {
+        None => summary_src,
+        Some(skip_list) => summary_src
+            .lines()
+            // We can unwrap because we already matched on the `skip_list` being a toml array
+            .filter(|haystack| {
+                let skip_list = skip_list.as_array().unwrap();
+                skip_list.iter().all(|needle| {
+                    // toml string arrays give you the opening and closing quotes - we need to trim them
+                    let needle = &needle.to_string();
```

Comment:
Something like this might save doing the 'contains' operation twice, and avoid some needless `std::to_string() -> String` operation.

```rust
let needle = needle.trim_matches('"');
if haystack.contains(needle) {
    log::info!("Skipped: {needle}");
    // don't want it
    false
} else {
    // do want it
    true
}
```

Reply:
Ah yes, that's much neater. Great tip.